scripts/Diffraction/isis_powder/osiris_routines/osiris_algs.py

```diff
@@ -228,23 +239,47 @@ def run_diffraction_focussing(run_number, drange_sets, calfile, van_norm=False,
     return focused
 
 
-def get_run_details(run_number_string, inst_settings, is_vanadium_run):
+def create_drange_sets(run_number_string, inst, file_ext):
+    run_number_list = common.generate_run_numbers(run_number_string=run_number_string)
+
+    drange_sets = {}
+    for run_number in run_number_list:
+        ws_name = inst._generate_input_file_name(run_number=run_number, file_ext=file_ext if file_ext else "")
+        ws = common.load_file(ws_name)
```

Comment:
So this function replaces the older `load_raw` function I think.
It seems like you could have written this so it resembles the `load_raw` where it calls common.load_raw_files and then you loop through the returned ws_list and call get_osiris_d_range on each? That would have meant you didn't need to change common.py and expose load_file?
Just trying to minimise unnecessary changes to the common code.

Reply:
Replaced